examples/diff.c

```diff
@@ -67,13 +75,70 @@ static int color_printer(
 	const git_diff_delta*, const git_diff_hunk*, const git_diff_line*, void*);
 static void diff_print_stats(git_diff *diff, struct opts *o);
 
+/*
+ * Read a file path
+ *
+ * path: a file path (char*)
+ *
+ * returns char* with the contents of the file (must be freed) or NULL if
+ * file does not exist or memory allocation failed
+ */
+char* read_file(const char* path) {
```

Comment:
I'd be more comfortable having that in "the example's common library". Reading files from a `char*` seems generally useful to example code.

Reply:
Will do ASAP. I'm impressed by your quick review, Yay!